packages/core-data/src/selectors.js

```diff
@@ -107,6 +107,20 @@ export function getEntityRecord( state, kind, name, key ) {
 	return get( state.entities.data, [ kind, name, 'queriedData', 'items', key ] );
 }
 
+/**
+ * Returns the Entity's record object by key. Doesn't trigger a resolver nor requests the entity from the API if the entity record isn't available in the local state.
+ *
+ * @param {Object} state  State tree
+ * @param {string} kind   Entity kind.
+ * @param {string} name   Entity name.
+ * @param {number} key    Record's key
+ *
+ * @return {Object?} Record.
+ */
+export function getEntityRecordNoResolver( state, kind, name, key ) {
```

Comment:
It bothers me that we're introducing this as a new API. For me resolvers are internal and shouldn't be known to public APIs

Reply:
By nature of the fact that there will always be cases where you want to run a selector locally/without resolvers, resolvers will always have to be somewhat "public."

We should have a `registry.selectLocal` or something like that, where the returned selectors don't trigger their resolvers.

Reply:
> By nature of the fact that there will always be cases where you want to run a selector locally/without resolvers

Why? The concept of resolvers is meant to be entirely transparent to the Component author. 
I'd argue that maybe it's needed for internal usage (inside the same store) but not as a public API of that store.

Reply:
Authors might also need to call selectors without triggering their resolvers, E.G., to get cached values and compare them with the new ones, to avoid API calls when there is no network, to keep local changes for things that could have been modified by other clients, etc.

Reply:
I'm not certain about that. The whole idea of the selectors/actions is that the component just asks for data and doesn't need to know whether it's async or sync (close to GraphQL approach of things).

Reply:
I'd tend to share the same concerns as @youknowriad, and I'd think as much as possible we should want to try to avoid it being a concern of a consumer to need to consider any existence of caching, network activity, etc. These are implementation details we probably want to consider as subject to future backwards-incompatible change, in the same way that we provide selectors and not the raw underlying state objects (since we want to be able to change state shapes).

In any case, is it something which is necessary to expose publicly in order to resolve the original issue? Doesn't seem like it would need to be. Even if it were a public API, I might be inclined to consider it more as an option to the existing selector, or something which could apply to _any_ selector that has associated resolver, rather than a new selector as controlling some characteristic of a specific existing selector. Or maybe we need to avoid triggering resolvers as a result of any `control` being yielded from an action.

I'm not sure if the documentation generators currently handle some concept of a "private" argument, but I could see this as something like a final object argument to `getEntityRecord`, which is evaluated in the resolver to decide whether to abort resolution.

```js
export function* getEntityRecord( state, kind, name, key, _options = {} ) {
	if ( _options.__unstableSkipResolution ) {
		return;
	}

	// ...
}
```

In fact, I wonder if that change alone would be fine, and just omit the last argument from the selector. The documentation generator won't consider the resolver, but I expect the `data` package will happily pass along all arguments anyways.

Reply:
>I'm not certain about that. The whole idea of the selectors/actions is that the component just asks for data and doesn't need to know whether it's async or sync (close to GraphQL approach of things).

GraphQL clients expose functions to run queries on the cache.

>something which could apply to any selector that has associated resolver

That's what I suggested with a `selectLocal` or `selectCached`.

>Or maybe we need to avoid triggering resolvers as a result of any control being yielded from an action.

That would force a lot of code duplication between controls and resolvers.

>In fact, I wonder if that change alone would be fine, and just omit the last argument from the selector. The documentation generator won't consider the resolver, but I expect the data package will happily pass along all arguments anyways.

Yes, that would also work, albeit it's a bit more manual than just having a `selectCached`. And would we leave it up to each resolver to decide what to do with the option or would we bail before even calling the resolver?

Reply:
Should we at least mark this s experimental before the beta release?

Reply:
https://github.com/WordPress/gutenberg/pull/20053

Reply:
I found this issue when reviewing and debugging another instance of this issue 🙂 

It seems to me that the root cause is that `getEntityRecords` (plural) fetches the records from network and stores them in local cache, but if I call `getEntityRecord` (singular) immediately after that, it's a cache miss. And fetches the record again. That shouldn't be happening. The `getEntityRecord` selector is effectively resolved, the cache contains up-to-date data and a new request shouldn't be fired. A GraphQL cache wouldn't behave this way.

Reply:
Yeah, I explained that further up. The problem is that this resolver setup doesn't have a trivial way to define relationships between resolvers like this. I guess we would need a new API like `getEntityRecord.shouldResolve = ( action, ...args ) => ...`.